### PR TITLE
Prepare redpanda chart to be processed by helm-docs

### PR DESCRIPTION
**Pull request by JakeSCahill**
This is part of https://github.com/redpanda-data/documentation/pull/1444.

- Added a template to customize the doc output of https://github.com/norwoodj/helm-docs
- Edited the descriptions in the redpanda `values.yaml` to be processed by https://github.com/norwoodj/helm-docs

I'll do the same for the console chart in a separate PR.

Example output in the docs preview: https://deploy-preview-1444--redpanda-documentation.netlify.app/docs/reference/redpanda-helm-spec/

### CODE REVIEW

**Comment by JakeSCahill** [comment]
All tests passed before except buildkite and /lint.

/lint complained about needing a version bump, so I did that. But, now other tests are failing too 😓

**Comment by joejulian** [comment]
It's failing yamllint on all the trailing spaces.

File: charts/redpanda/values.yaml [hunk 1]
```diff
@@ -495,33 +533,33 @@ statefulset:
...
-  # Increases the maximum number of outstanding asynchronous IO operations if the
-  # current value is below a certain threshold. This allows redpanda to make as many
+  # -- Increase the maximum number of outstanding asynchronous IO operations if the
+  # current value is below a certain threshold. This allows Redpanda to make as many
   # simultaneous IO requests as possible, increasing throughput.
   #
-  # Enabling this option will create a privileged container. If your security profile does not allow this,
-  # see https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-tune-workers/ for tuning requirements.
+  # When this option is enabled, Helm creates a privileged container. If your security profile does not allow this,
+  # For tuning requirements,
```
Comment (JakeSCahill) [inline charts/redpanda/values.yaml:561]:
```suggestion
```

**Comment by alejandroEsc** [comment]
This is annoying, but we will need to bump the chart version it seems?

File: charts/redpanda/Chart.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ type: application
 # The chart version and the app version are not the same and will not track
 # together. The chart version is a semver representation of changes to this
 # chart.
-version: 3.0.7
+version: 3.0.8
```
Comment (JakeSCahill) [inline charts/redpanda/Chart.yaml:26]:
```suggestion
version: 3.0.9
```

**Comment by joejulian** [comment]
Actually, there's no need to bump the chart version. This should be excluded like markdown files are here: https://github.com/redpanda-data/helm-charts/blob/main/.github/workflows/pull_requests.yaml#L28

We should also add a .helmignore file so we don't package this when the chart is published.

File: charts/redpanda/Chart.yaml [hunk 3]
```diff
@@ -23,7 +23,7 @@ type: application
 # The chart version and the app version are not the same and will not track
 # together. The chart version is a semver representation of changes to this
 # chart.
-version: 3.0.8
+version: 3.0.9
```
Comment (joejulian) [inline charts/redpanda/Chart.yaml:26]:
```suggestion
version: 3.0.8
```

**Review by joejulian — APPROVED** [review]

**Pull request merged**